tests/test_configuration_class.py

```diff
@@ -80,6 +80,91 @@ def test_config_from_cli():
     assert len(scenario_mgr) == 1
 
 
+def test_config_from_scenarios():
+    user_dict = {
+        "input_model": "reeds-US",
+        "output_model": "sienna",
+        "scenarios": [
+            {
+                "name": "test2030",
+                "weather_year": 2015,
+                "solve_year": 2030,
+            },
+            {
+                "name": "test2050",
+                "weather_year": 2015,
+                "solve_year": 2055,
+            },
+        ],
+    }
+
+    config = Configuration.from_scenarios({}, user_dict)
+    assert config is not None
+    assert len(config) == 2
+    assert config["test2030"]
+    assert config["test2030"].solve_year == 2030
+
+
+@pytest.mark.parametrize(
+    "cli_input,user_dict",
+    [
+        (
+            {
+                "name": "Test",
+                "weather_year": 2015,
+                "solve_year": [2055],
+                "input_model": "plexos",
+                "output_model": "sienna",
+            },
+            {},
+        ),
+        (
+            {},
+            {
+                "input_model": "reeds-US",
+                "output_model": "sienna",
+                "scenarios": [
+                    {
+                        "name": "test2030",
+                        "weather_year": 2015,
+                        "solve_year": 2030,
+                    },
+                    {
+                        "name": "test2050",
+                        "weather_year": 2015,
+                        "solve_year": 2055,
+                    },
+                ],
+            },
+        ),
+        (
+            {
+                "name": "Test",
+                "weather_year": 2015,
+                "input_model": "plexos",
+                "output_model": "sienna",
+            },
+            {
+                "scenarios": [
+                    {
+                        "name": "test2030",
+                        "solve_year": 2030,
+                    },
+                    {
+                        "name": "test2050",
+                        "solve_year": 2055,
+                    },
+                ],
+            },
+        ),
+    ],
+    ids=["no-user-dict", "no-cli", "both"],
+)
+def test_get_config(cli_input, user_dict):
```

Comment:
would it be worth adding a component of this test that specifies the correct overrides given cli and user_dict input?

Reply:
Done.